src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +842,348 @@ def __try_add_error(error_list, detail):
+                    os.remove(file)
+                    files_to_removed.append(file)
+            except Exception as e:
+                self.composite_logger.log_debug("Error deleting complete status file. [File={0} [Exception={1}]]".format(repr(file), repr(e)))
+
+        self.composite_logger.log_debug("Cleaned up older complete status files: {0}".format(files_to_removed))
+
+    def __calc_status_size_on_disk(self, full_status):
+        """ Calculate status file size in bytes on disk """
+        return len(json.dumps(full_status).encode("utf-8"))
+
+    def __calc_package_payload_size_on_disk(self, package_list):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def size_of_constant_status_data(self, complete_status_file_payload, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without package data  """
+        status_file_no_list_data = complete_status_file_payload
+        if assessment_status_index is not None:
+            assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(status_file_no_list_data, assessment_status_index)
+            assessment_msg_empty_list = self.__update_substatus_msg_packages_and_errors(assessment_substatus_msg_copy, [], assessment_substatus_msg_copy['errors'])
+            status_file_no_list_data['status']['substatus'][assessment_status_index]['formattedMessage']['message'] = json.dumps(assessment_msg_empty_list)
+
+        if installation_status_index is not None:
+            installation_substatus_msg_copy = self.__get_substatus_msg_at_index(status_file_no_list_data, installation_status_index)
+            installation_msg_empty_list = self.__update_substatus_msg_packages_and_errors(installation_substatus_msg_copy, [], installation_substatus_msg_copy['errors'])
+            status_file_no_list_data['status']['substatus'][installation_status_index]['formattedMessage']['message'] = json.dumps(installation_msg_empty_list)
+
+        return self.__calc_status_size_on_disk(status_file_no_list_data)
+
+    def __get_substatus_index(self, substatus_name, substatus):
+        """" Get substatus index from the current substatus """
+        for index, sub_name in enumerate(substatus):
+            if sub_name['name'] == substatus_name:
+                return index
+        return None
+
+    def __recompose_truncated_status_file(self, truncated_status_file, truncated_package_list, count_total_errors, substatus_json, substatus_name, substatus_index):
+        """ Recompose final truncated status file version """
+        truncated_detail_list = []
+        error_code = Constants.PatchOperationTopLevelErrorCode.ERROR
+        code, errors_details = self.__get_current_complete_status_errors(substatus_json)
+
+        # Check for existing assessment errors before recompose
```

Comment:
Check for existing substatus errors before recomposing

This is not specific to assessment now

Reply:
comment removed